libs/sdk/src/react/utils.ts

```diff
@@ -0,0 +1,96 @@
+import type {
+  Message,
+  AIMessage,
+  DefaultToolCall,
+  UIMessage,
+} from "../types.messages.js";
+import { getReasoningMessage } from "../utils/reasoning.js";
+
+/**
+ * Checks if an AI message has non-reasoning text content.
+ * Returns true if the message has string content or text blocks in array content.
+ */
+function hasNonReasoningContent(message: AIMessage): boolean {
+  const { content } = message;
+
+  // String content
+  if (typeof content === "string") {
+    return content.trim().length > 0;
+  }
+
+  // Array content - check for text blocks
+  if (Array.isArray(content)) {
+    for (const block of content) {
+      if (typeof block === "object" && block !== null && "type" in block) {
+        const typed = block as { type: string; text?: string };
+        // Text block with actual content
+        if (
+          typed.type === "text" &&
+          typeof typed.text === "string" &&
+          typed.text.trim().length > 0
+        ) {
+          return true;
+        }
+      }
+    }
+  }
```

Comment:
I think as part of SDK, we should coerce `string` content to text blocks?

Reply:
Should we tackle this in a separate PR?